### PR TITLE
[Ready for Review]Add multi-label text classification example

### PR DESCRIPTION
**Pull request by linjieccc**
### PR types
 Others

### PR changes
Others

### Description
add multi-label text classification example

### CODE REVIEW

File: examples/text_classification/multi_label/data.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
```
Comment (Steffy-zxf) [inline examples/text_classification/multi_label/data.py:1]: 建议集成为paddlenlp内置数据集。
直接放在 paddlenlp/datasets/toxic.py

File: examples/text_classification/multi_label/train.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+def do_train():
+    paddle.set_device(args.device)
+    rank = paddle.distributed.get_rank()
+    if paddle.distributed.get_world_size() > 1:
+        paddle.distributed.init_parallel_env()
+
+    set_seed(args.seed)
+
+    # Load datasets.
+    dataset_name = 'train.csv'
```
Comment (Steffy-zxf) [inline examples/text_classification/multi_label/train.py]: 改成
`load_dataset("toxic", splits=['train', 'test'])`

File: examples/text_classification/multi_label/train.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    train_ds = load_dataset(read_custom_data, filename=os.path.join(args.data_path, dataset_name), is_test=False, lazy=False)
+
+    # If you wanna use bert/roberta/electra pretrained model,
+    model = ppnlp.transformers.BertForMultiLabelTextClassification.from_pretrained(
```
Comment (Steffy-zxf) [inline examples/text_classification/multi_label/train.py]:
```shell
from paddlenlp.transformers import BertForMultiLabelTextClassification

model = BertForMultiLabelTextClassification.from_pretrained('bert-base-uncased', num_labels=...)
```

File: examples/text_classification/multi_label/metric.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                else:
+                    self.fp[y_pred] = self.fp.get(y_pred, 0) + 1
+                    self.fn[y] = self.fn.get(y, 0) + 1
+
+    def accumulate(self):
+        """
+        Calculate the final micro F1 score.
+        Returns:
+            A scaler float: results of the calculated micro F1 score.
+        """
```
Comment (Steffy-zxf) [inline examples/text_classification/multi_label/metric.py]: 具体计算可以直接调用sklearn.metric 计算 多标签分类指标

**Review by Steffy-zxf — CHANGES_REQUESTED** [review]
数据集建议改成直接集成为paddlenlp内部数据集

File: examples/text_classification/multi_label/data.py (hunk 1 above)
Comment (Steffy-zxf) [reply examples/text_classification/multi_label/data.py:1]: 由于toxic数据集需要注册kaggle，才能下载。所以还是采用通过自定义数据集方式加载数据。

File: examples/text_classification/multi_label/metric.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
```
Comment (Steffy-zxf) [inline examples/text_classification/multi_label/metric.py:1]: 再增加一个auc指标统计，计算每个tag的auc分数。

File: examples/text_classification/multi_label/predict.py [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+        results(obj:`dict`): All the predictions labels.
+    """
+
+    results = []
+    model.eval()
+    for step, batch in enumerate(data_loader, start=1):
+        input_ids, token_type_ids = batch
+        logits = model(input_ids, token_type_ids)
+        probs = F.sigmoid(logits)
+        print(probs)
```
Comment (Steffy-zxf) [inline examples/text_classification/multi_label/predict.py]: 删除多余的空行

File: examples/text_classification/multi_label/predict.py [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+    """write_results"""
+    cols = ["id", "toxic", "severe_toxic", "obscene", "threat", "insult", "identity_hate"]
+    data = pd.read_csv(filename)
+    qids = [line[0] for line in data.values]
+    results_dict["id"] = qids
+    results = list(map(list, zip(*results)))
+    for key in results_dict:
+        if key != "id":
+            for result in results:
+                results_dict[key] = result
```
Comment (Steffy-zxf) [inline examples/text_classification/multi_label/predict.py]: 建议增加一个threshhold参数，控制最终label的映射。

File: examples/text_classification/multi_label/predict.py [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+        if step % 100 == 0:
+            print("step %d, %d samples processed" % (step, step * batch_size))
+    return results
+
+
+if __name__ == "__main__":
+    paddle.set_device(args.device)
+
+    # Load train dataset.
+    dataset_name = 'test.csv'
```
Comment (Steffy-zxf) [inline examples/text_classification/multi_label/predict.py]: dataset_name -> file_name

File: examples/text_classification/multi_label/predict.py [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+    "Sequences longer than this will be truncated, sequences shorter will be padded.")
+parser.add_argument("--batch_size", default=64, type=int, help="Batch size per GPU/CPU for training.")
+parser.add_argument('--device', choices=['cpu', 'gpu', 'xpu'], default="gpu", help="Select which device to train model, defaults to gpu.")
+parser.add_argument("--data_path", type=str, default="./data", help="The path of datasets to be loaded")
+args = parser.parse_args()
+# yapf: enable
+
+def write_results(filename, results, results_dict):
+    """write_results"""
+    cols = ["id", "toxic", "severe_toxic", "obscene", "threat", "insult", "identity_hate"]
```
Comment (Steffy-zxf) [inline examples/text_classification/multi_label/predict.py]: `cols` 和 `results_dict` 作用应该是相同的，可以去掉`results_dict`

File: examples/text_classification/multi_label/predict.py [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+
+if __name__ == "__main__":
+    paddle.set_device(args.device)
+
+    # Load train dataset.
+    dataset_name = 'test.csv'
+    test_ds = load_dataset(read_custom_data, filename=os.path.join(args.data_path, dataset_name), is_test=True, lazy=False)
+
+    # Init the results template.
+    results_dict = {"toxic": [], "severe_toxic": [], "obscene": [], "threat": [], "insult": [], "identity_hate": []}
```
Comment (Steffy-zxf) [inline examples/text_classification/multi_label/predict.py]: `results_dict`应该不需要

File: examples/text_classification/multi_label/predict.py [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+
+    # If you wanna use bert/roberta/electra pretrained model,
+    model = ppnlp.transformers.BertForMultiLabelTextClassification.from_pretrained(
```
Comment (Steffy-zxf) [inline examples/text_classification/multi_label/predict.py]: from paddlenlp.transformers import BertForMultiLabelTextClassification, BertTokenzier

model = BertForMultiLabelTextClassification.from_pretrained('bert-base-uncased', num_labels=...)
tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
```

File: examples/text_classification/multi_label/predict.py [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+        'bert-base-uncased', num_labels=len(results_dict))
+
+    # If you wanna use bert/roberta/electra pretrained model,
```
Comment (Steffy-zxf) [inline examples/text_classification/multi_label/predict.py]: 相应注释注意修改

File: examples/text_classification/multi_label/train.py [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+        Stack()  # label
+    ): [data for data in fn(samples)]
+    train_data_loader = create_dataloader(
+        train_ds,
+        mode='train',
+        batch_size=args.batch_size,
+        batchify_fn=batchify_fn,
+        trans_fn=trans_func)
+    if args.do_eval:
+        dataset_name = 'valid.csv'
```
Comment (Steffy-zxf) [inline examples/text_classification/multi_label/train.py]: toxic 没有验证集，保持和原始数据集一致。删去验证集。

File: paddlenlp/transformers/bert/modeling.py [hunk 14]
```diff
@@ -508,6 +509,41 @@ def forward(self,
         logits = self.classifier(sequence_output)
         return logits
 
+class BertForMultiLabelTextClassification(BertPretrainedModel):
+    """
+    Model for multi label text classification task with BERT.
+    Args:
+        bert (BertModel): An instance of BertModel.
```
Comment (Steffy-zxf) [inline paddlenlp/transformers/bert/modeling.py]: bert (BertModel, nn.Layer): An instance of BertModel.

**Review by Steffy-zxf — CHANGES_REQUESTED** [review]
增加auc评价分数统计

File: examples/text_classification/multi_label/train.py (hunk 3 above)
Comment (linjieccc) [reply examples/text_classification/multi_label/train.py]: done

File: examples/text_classification/multi_label/metric.py (hunk 4 above)
Comment (linjieccc) [reply examples/text_classification/multi_label/metric.py]: done
修改为sklearn的f1 score和 auc

File: examples/text_classification/multi_label/predict.py (hunk 6 above)
Comment (linjieccc) [reply examples/text_classification/multi_label/predict.py]: done

File: examples/text_classification/multi_label/predict.py (hunk 7 above)
Comment (linjieccc) [reply examples/text_classification/multi_label/predict.py]: done

File: examples/text_classification/multi_label/predict.py (hunk 8 above)
Comment (linjieccc) [reply examples/text_classification/multi_label/predict.py]: done

File: examples/text_classification/multi_label/predict.py (hunk 9 above)
Comment (linjieccc) [reply examples/text_classification/multi_label/predict.py]: results_dict在write csv的时候要用到，去掉了cols

File: examples/text_classification/multi_label/predict.py (hunk 10 above)
Comment (linjieccc) [reply examples/text_classification/multi_label/predict.py]: 已改为label_info，更直观些

File: examples/text_classification/multi_label/predict.py (hunk 11 above)
Comment (linjieccc) [reply examples/text_classification/multi_label/predict.py]: done

File: examples/text_classification/multi_label/predict.py (hunk 12 above)
Comment (linjieccc) [reply examples/text_classification/multi_label/predict.py]: done

File: examples/text_classification/multi_label/train.py (hunk 13 above)
Comment (linjieccc) [reply examples/text_classification/multi_label/train.py]: done

File: paddlenlp/transformers/bert/modeling.py (hunk 14 above)
Comment (linjieccc) [reply paddlenlp/transformers/bert/modeling.py]: done

File: examples/text_classification/multi_label/README.md [hunk 15]
```diff
@@ -0,0 +1,156 @@
+# 使用预训练模型Fine-tune完成多标签文本分类任务
```
Comment (ZeyuChen) [inline examples/text_classification/multi_label/README.md]: 建议使用简单的文本来做标题，Q3会逐步国际化，所以中文文档的措辞要尽可能先用英文思维来考虑。

File: examples/text_classification/multi_label/README.md [hunk 16]
```diff
@@ -0,0 +1,156 @@
...
+
+在2017年之前，工业界和学术界对NLP文本处理依赖于序列模型[Recurrent Neural Network (RNN)](../rnn).
+
+<p align="center">
+<img src="http://colah.github.io/posts/2015-09-NN-Types-FP/img/RNN-general.png" width="40%" height="30%"> <br />
+</p>
+
+
+[paddlenlp.seq2vec是什么? 瞧瞧它怎么完成情感分析](https://aistudio.baidu.com/aistudio/projectdetail/1283423)教程介绍了如何使用`paddlenlp.seq2vec`表征文本语义。
+
```
Comment (Steffy-zxf) [inline examples/text_classification/multi_label/README.md:2]: 以上内容可以删去。

File: examples/text_classification/multi_label/README.md [hunk 17]
```diff
@@ -0,0 +1,156 @@
...
+从Kaggle下载[Toxic Comment Classification Challenge](https://www.kaggle.com/c/jigsaw-toxic-comment-classification-challenge/data)数据集并将数据集文件放在`./data`路径下。
+以下是`./data`路径的文件组成：
+
+```text
+data/
+├── sample_submission.csv
+├── train.csv
+├── test.csv
+└── test_labels.csv
```
Comment (Steffy-zxf) [inline examples/text_classification/multi_label/README.md]: test_labels.csv 是啥？注明下每个文件的含义。

File: paddlenlp/transformers/bert/modeling.py [hunk 18]
```diff
@@ -29,6 +29,7 @@
     'BertForSequenceClassification',
     'BertForTokenClassification',
     'BertForQuestionAnswering',
+    'BertForMultiLabelTextClassification',
```
Comment (ZeyuChen) [inline paddlenlp/transformers/bert/modeling.py]: 不建议使用这个Text这个命名，请横向对标下HF，看看HF那边MultipleChoice这个任务是否可以对上。

**Review by ZeyuChen — COMMENTED** [review]
BertForMultiLabelTextClassification
不太建议新增这个Bert任务。这样的话得库里所有任务都匹配。建议使用自定义任务的方式来呈现。或者看看MultipleChoice这个任务对不对得上。

File: examples/text_classification/multi_label/train.py (hunk 2 above)
Comment (Steffy-zxf) [reply examples/text_classification/multi_label/train.py]: 由于toxic数据集需要注册kaggle，才能下载。所以还是采用通过自定义数据集方式加载数据。

File: examples/text_classification/multi_label/predict.py [hunk 19]
```diff
@@ -0,0 +1,106 @@
...
+
+
+if __name__ == "__main__":
+    paddle.set_device(args.device)
+
+    # Load train dataset.
+    file_name = 'test.csv'
+    test_ds = load_dataset(read_custom_data, filename=os.path.join(args.data_path, file_name), is_test=True, lazy=False)
+
+    # Init the results template.
```
Comment (Steffy-zxf) [inline examples/text_classification/multi_label/predict.py]: The dataset labels

File: examples/text_classification/multi_label/predict.py [hunk 20]
```diff
@@ -0,0 +1,106 @@
...
+    label_info = ['toxic', 'severe_toxic', 'obscene', 'threat', 'insult', 'identity_hate']
+
+    # Init bert pretrained model
```
Comment (Steffy-zxf) [inline examples/text_classification/multi_label/predict.py]: Load model.

File: examples/text_classification/multi_label/predict.py [hunk 21]
```diff
@@ -0,0 +1,106 @@
...
+    model = BertForMultiLabelTextClassification.from_pretrained(
+        'bert-base-uncased', num_labels=len(label_info))
+
+    # Init bert tokenizer
```
Comment (Steffy-zxf) [inline examples/text_classification/multi_label/predict.py]: Load Bert Tokenizer

File: examples/text_classification/multi_label/README.md (hunk 16 above)
Comment (linjieccc) [reply examples/text_classification/multi_label/README.md:2]: done

File: examples/text_classification/multi_label/README.md (hunk 15 above)
Comment (linjieccc) [reply examples/text_classification/multi_label/README.md]: done

File: paddlenlp/transformers/bert/modeling.py (hunk 18 above)
Comment (linjieccc) [reply paddlenlp/transformers/bert/modeling.py]: 已修改为自定义任务形式组网

File: examples/text_classification/multi_label/README.md (hunk 17 above)
Comment (linjieccc) [reply examples/text_classification/multi_label/README.md]: done

File: examples/text_classification/multi_label/model.py [hunk 22]
```diff
@@ -0,0 +1,52 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import paddle.nn as nn
+from paddlenlp.transformers import BertPretrainedModel
+
+class BertForMultiLabelClassifier(BertPretrainedModel):
```
Comment (Steffy-zxf) [inline examples/text_classification/multi_label/model.py]: 实际上不仅可以用bert，还能使用ernie、roberta等transformer模型。
建议改成 `class MultiLabelClassifier(nn.Layer)`，transformer模型作为一个sublayer。

**Review by Steffy-zxf — APPROVED** [review]
LGTM
